dali/test/python/test_operator_spectrogram.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import nvidia.dali as dali
+import numpy as np
+from numpy.testing import assert_array_equal, assert_allclose
+from functools import partial
+from test_utils import check_batch
+from test_utils import compare_pipelines
+from test_utils import RandomDataIterator
+from test_utils import DataIterator
+import librosa as librosa
+import math
+
+class SpectrogramPipeline(Pipeline):
+    def __init__(self, device, batch_size, iterator, nfft, window_length, window_step,
+                 num_threads=1, device_id=0):
+        super(SpectrogramPipeline, self).__init__(batch_size, num_threads, device_id)
+        self.device = device
+        self.iterator = iterator
+        self.inputs = ops.ExternalSource()
+        self.fft = ops.Spectrogram(device = self.device,
+                                   nfft = nfft,
+                                   window_length = window_length,
+                                   window_step = window_step,
+                                   power = 2)
+
+    def define_graph(self):
+        self.data = self.inputs()
+        out = self.data.gpu() if self.device == 'gpu' else self.data
+        out = self.fft(out)
+        return out
+
+    def iter_setup(self):
+        data = self.iterator.next()
+        self.feed_input(self.data, data)
+
+def spectrogram_func_torch(nfft, win_len, win_step, input):
+    def hann_win(n):
```

Comment:
Can we have this `hann_win` extracted to one place. Now I see it twice.

Reply:
This torch implementation was meant to be replaces by librosa (there is no need to check both as torchaudio has already test verifying their implementation against librosa).